orb-ui/examples/ui-replay.rs

```diff
@@ -47,37 +79,40 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         )
         .init();
 
+    let args = Args::parse();
     let connection = Connection::session().await?;
     let proxy = SignupStateProxy::new(&connection).await?;
 
     // set initial state
-    let _ = proxy.orb_signup_state_event(format!("\"Bootup\"")).await;
-    let _ = proxy.orb_signup_state_event(format!("\"Idle\"")).await;
+    let _ = proxy.orb_signup_state_event("\"Bootup\"".to_string()).await;
+    let _ = proxy.orb_signup_state_event("\"Idle\"".to_string()).await;
     let _ = proxy
-        .orb_signup_state_event(format!("\"SoundVolume {{ level: 10 }}\""))
+        .orb_signup_state_event("\"SoundVolume {{ level: 10 }}\"".to_string())
         .await;
 
     // get path to records file from program arguments or use default
-    let path = std::env::args().nth(1).unwrap_or("records.txt".to_string());
-    let file = File::open(path)?;
+    let path = args.path.unwrap_or(RECORDS_FILE.to_string());
+    let file = File::open(path.clone()).wrap_err(format!("cannot open {path}"))?;
```

Comment:
nit: if the error type is expensive to create (such as allocating a string), then you want to lazily construct it. `wrap_err` constructs it eagerly, whereas `wrap_err_with` does this lazily with a closure that gets invoked *only* on the error case.
```suggestion
    let file = File::open(path.clone()).wrap_err_with(|| format!("cannot open {path}"))?;
```